tap_s3_csv/__init__.py

```diff
@@ -81,8 +82,11 @@ def main() -> None:
     args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS)
     config = args.config
 
-    # Reassign the config tables to the validated object
-    config['tables'] = CONFIG_CONTRACT(config.get('tables', {}))
+    # Reassign the config tables to the validated object, accept encoded json
```

Comment:
what's the use case here? why isn't the current implement not sufficient?

Reply:
To preface, I'm running this tap manually outside of pipeline wise. A common pattern which we use is to store our singer configurations as environment variables and right before execution we write them out to the config.json. In this case our tables key needs to be serialized as a string. I believe the meltano project does something similar with their configs. I was hoping with my changes to just support both json or a serialized json string.

Reply:
I actually decided to handle this on my end - It seems like its better to leave the tap using a dict instead of also supporting a string. I'll close the PR. Thanks for the feedback